### PR TITLE
fixed bug in enrichment testing

### PR DESCRIPTION
**Pull request by wxli0**
relates to #481 point 4
- enrichment.js, use object instead of map
- enrichment-test.js, 't depth', whitespaces in 't name' also change;

### CODE REVIEW

File: test/server/enrichment-map/enrichment/enrichment-test.js [hunk 1]
```diff
@@ -1,92 +1,101 @@
...
 
 // object keys are unordered
 // t group changes
 const objectEquality = (obj1, obj2) => {
   return _.isEqualWith(obj1, obj2, (val1, val2, key) => {
-      if (key === "signf" || key === "pvalue" || key === "T" || key === "Q" || key === "Q&T" || key === "Q&T/Q" || key === "Q&T/T" || key === "t type" || key === "t name" || key === "t depth" || key === "Q&T list") {
-        return val1.key === val2.key;
-      }
-      return true;
+    if (key === 't group' || key === 't depth') { return true; }
```
Comment (d2fong) [inline test/server/enrichment-map/enrichment/enrichment-test.js]: Add a comment saying why these keys are ignored specifically

File: test/server/enrichment-map/enrichment/enrichment-test.js [hunk 2]
```diff
@@ -1,90 +1,102 @@
...
+            'Q&T/Q': '1.000',
+            'Q&T/T': '0.012',
+            't type': 'rea',
+            't group': '2',
+            't name': '   RNA Polymerase II Pre-transcription Events',
+            't depth': '1',
+            'Q&T list': 'AFF4'
+          }
+      };
+      expect(objectEquality(result, res)).to.equal(true);
```
Comment (maxkfranz) [inline test/server/enrichment-map/enrichment/enrichment-test.js]: Please read the chai docs:  http://chaijs.com/api/bdd/

`expect(foo).to.deep.equal(bar)`

Don't re-implement things that have been done in libraries.

File: test/server/enrichment-map/enrichment/enrichment-test.js [hunk 3]
```diff
@@ -1,90 +1,102 @@
...
+            T: '83',
+            Q: '1',
+            'Q&T': '1',
+            'Q&T/Q': '1.000',
+            'Q&T/T': '0.012',
+            't type': 'rea',
+            't group': '2',
+            't name': '   RNA Polymerase II Pre-transcription Events',
+            't depth': '1',
+            'Q&T list': 'AFF4'
```
Comment (maxkfranz) [inline test/server/enrichment-map/enrichment/enrichment-test.js]: Is this the expected format of the response?  These are not proper variable names...

Comment (wxli0) [reply test/server/enrichment-map/enrichment/enrichment-test.js]: We only check for specific keys here. renamed the function to 'gProfilerResEquality'.

File: test/server/enrichment-map/enrichment/enrichment-test.js (hunk 2 above)
Comment (wxli0) [reply test/server/enrichment-map/enrichment/enrichment-test.js]: We only check for specific keys here. renamed the function to 'gProfilerResEquality'.

**Pull request merged**